File: deploy_fix.py
import paramiko
import os

def run():
    client = paramiko.SSHClient()
    client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
    client.connect('62.72.32.37', username='root', password='Aa7161062.123')
    
    with client.open_sftp() as sftp:
        # Backend
        sftp.put(r'd:\Complete\ACSES Scholar\Quantum uz backend\application\models.py', '/var/www/acses_backend/application/models.py')
        sftp.put(r'd:\Complete\ACSES Scholar\Quantum uz backend\application\serializers.py', '/var/www/acses_backend/application/serializers.py')
        print("Uploaded models/serializers")
        
        sftp.put(r'd:\Complete\ACSES Scholar\seed.py', '/var/www/acses_backend/seed.py')
        print("Uploaded seed.py")
        
        # Frontend
        frontend_base = r'd:\Complete\ACSES Scholar'
        remote_base = '/var/www/acses_scholar'
        
        # Files to upload
        files = [
            'lib/catalog.ts',
            'app/components/features/LessonPlayer.tsx',
            'app/(site)/classes/page.tsx',
            'app/(site)/classes/[classId]/page.tsx',
            'app/(site)/classes/[classId]/[subjectId]/page.tsx',
            'app/(site)/classes/[classId]/[subjectId]/[lessonId]/page.tsx'
        ]
        
        for f in files:
            local_path = os.path.join(frontend_base, f.replace('/', os.sep))
            remote_path = remote_base + '/' + f
            # Ensure remote directory exists
            remote_dir = '/'.join(remote_path.split('/')[:-1])
            client.exec_command(f"mkdir -p {remote_dir}")
            sftp.put(local_path, remote_path)
            print(f"Uploaded {f}")
            
    # Run migrations
    print("Running migrations...")
    stdin, stdout, stderr = client.exec_command('cd /var/www/acses_backend && ./venv/bin/python3 manage.py makemigrations && ./venv/bin/python3 manage.py migrate')
    print(stdout.read().decode('utf-8', errors='ignore'))
    
    # Run seed
    print("Running seed script...")
    stdin, stdout, stderr = client.exec_command('cd /var/www/acses_backend && ./venv/bin/python3 seed.py')
    print(stdout.read().decode('utf-8', errors='ignore'))
    print(stderr.read().decode('utf-8', errors='ignore'))
    
    # Restart Backend
    print("Restarting backend...")
    client.exec_command('systemctl restart acses_backend')
    
    # Rebuild Frontend
    print("Rebuilding frontend...")
    stdin, stdout, stderr = client.exec_command('cd /var/www/acses_scholar && npm run build && pm2 restart acses_frontend')
    print(stdout.read().decode('utf-8', errors='ignore'))
    print(stderr.read().decode('utf-8', errors='ignore'))
    
    print("All done!")
    client.close()

if __name__ == "__main__":
    run()

File: deploy_fix.js
const { NodeSSH } = require('node-ssh');
const path = require('path');
const fs = require('fs');

const ssh = new NodeSSH();

async function run() {
  try {
    await ssh.connect({
      host: '62.72.32.37',
      username: 'root',
      password: 'Aa7161062.123'
    });

    // 1. Upload Backend Files
    await ssh.putFile('d:\\Complete\\ACSES Scholar\\Quantum uz backend\\application\\models.py', '/var/www/acses_backend/application/models.py');
    await ssh.putFile('d:\\Complete\\ACSES Scholar\\Quantum uz backend\\application\\serializers.py', '/var/www/acses_backend/application/serializers.py');
    await ssh.putFile('d:\\Complete\\ACSES Scholar\\seed.py', '/var/www/acses_backend/seed.py');
    console.log("Uploaded Backend Files");

    // 2. Upload Frontend Files
    const frontendBase = 'd:\\Complete\\ACSES Scholar';
    const remoteBase = '/var/www/acses_scholar';
    const files = [
      'lib/catalog.ts',
      'app/components/features/LessonPlayer.tsx',
      'app/(site)/classes/page.tsx',
      'app/(site)/classes/[classId]/page.tsx',
      'app/(site)/classes/[classId]/[subjectId]/page.tsx',
      'app/(site)/classes/[classId]/[subjectId]/[lessonId]/page.tsx'
    ];

    for (const f of files) {
      const localPath = path.join(frontendBase, f);
      const remotePath = remoteBase + '/' + f;
      const remoteDir = remotePath.split('/').slice(0, -1).join('/');
      await ssh.execCommand(`mkdir -p ${remoteDir}`);
      await ssh.putFile(localPath, remotePath);
      console.log(`Uploaded ${f}`);
    }

    // 3. Migrations
    console.log("Running migrations...");
    const migResult = await ssh.execCommand('./venv/bin/python3 manage.py makemigrations && ./venv/bin/python3 manage.py migrate', { cwd: '/var/www/acses_backend' });
    console.log(migResult.stdout || migResult.stderr);

    // 4. Seed
    console.log("Running seed...");
    const seedResult = await ssh.execCommand('./venv/bin/python3 seed.py', { cwd: '/var/www/acses_backend' });
    console.log(seedResult.stdout || seedResult.stderr);

    // 5. Restart Backend
    console.log("Restarting Backend...");
    await ssh.execCommand('systemctl restart acses_backend');

    // 6. Rebuild Frontend
    console.log("Rebuilding Frontend (this takes a while)...");
    const buildResult = await ssh.execCommand('npm run build && pm2 restart acses_frontend', { cwd: '/var/www/acses_scholar' });
    console.log(buildResult.stdout || buildResult.stderr);

    console.log("All tasks completed!");
    process.exit(0);
  } catch (err) {
    console.error(err);
    process.exit(1);
  }
}

run();
